GetStream/stream-chat-android
# [3176] Apply fixed image and link dimensions in order to avoid items jumping upon loading

### 🎯 Goal

Closes #3176 

The second iteration of our CDN provides us with fixed sizes for images. This allows us to prevent jumping images changing their sizes abruptly while loading, which had previously lead to focused items such as when navigating to original messages the reply was made to jumping out of focus.

### 🛠 Implementation details

- Eagerly set the dimension ratio and dimension of the image attachment containers in `ImageAttachmentGroupView` if we are displaying a single image
- Set a fixed size (dimen ratio) for the link previews containers

### 🎨 UI Changes

_Add relevant screenshots_

| Before | After |
| --- | --- |
| ![before](https://user-images.githubusercontent.com/37080097/183652615-caa71b8e-cd54-4d15-9fb8-19a2f9a5493f.gif) | ![after](https://user-images.githubusercontent.com/37080097/183652553-1972ea1f-462d-49cc-a6a9-a66c856cfe1a.gif) |


### 🧪 Testing

Repeat the same steps for develop and this current branch
**Important:** Uninstall and reinstall the app or flush the storage & cache before switching branches. Otherwise images might remain loaded in the memory and the results might get muddied. 

1. Log in as Filip, Amit, Jc, Thierry or Tommasso
2. Enter channels `Fixed image link test` and `Fixed image sizes test channel`
3. Scroll upwards and view the links/ images

Result: The images and links should "jump" on `develop` but not on the PR branch.

1. Log in as Filip, Amit, Jc, Thierry or Tommasso
2. Enter channel `WS Events
3. Using the message `Use this for testing` sent by JC, navigate to the original message the reply was made to

Result: On `develop`, the message should be bumped out of focus by loading neighboring images/ links. On the PR branch, it should remain in focus.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

[!gif](https://media0.giphy.com/media/fasbtcqFYmzOcb2fjA/giphy.gif?cid=ecf05e477xccprttxv059z18vjeg4mofzwv1pjlh190o9hof&rid=giphy.gif&ct=g)
